### ISSUE TITLE
2022.08.15 Not starting. Cannot set verify_mode to CERT_NONE and no attribute 'NO_LGMARGIN

### ISSUE THREAD

**Issue by jester-xbmc** [SickChill/SickChill#8068]
running docker image on DSM 7.1.1 (NAS) 
after hitting update yesterday, crashes with the below:

Traceback (most recent call last):
  File "/app/sickchill/SickChill.py", line 350, in <module>
    main()
  File "/app/sickchill/SickChill.py", line 345, in main
    SickChill().start()
  File "/app/sickchill/SickChill.py", line 151, in start
    sickchill.start.initialize(consoleLogging=self.console_logging)
  File "/app/sickchill/sickchill/start.py", line 214, in initialize
    helpers.set_opener(settings.SSL_VERIFY)
  File "/app/sickchill/sickchill/oldbeard/helpers.py", line 71, in set_opener
    https_handler = HTTPSHandler(context=make_context(verify), check_hostname=True)
  File "/app/sickchill/sickchill/oldbeard/helpers.py", line 62, in make_context
    context.verify_mode = ssl.CERT_REQUIRED if verify else ssl.CERT_NONE
  File "/usr/local/lib/python3.9/ssl.py", line 720, in verify_mode
    super(SSLContext, SSLContext).verify_mode.__set__(self, value)
ValueError: Cannot set verify_mode to CERT_NONE when check_hostname is enabled.

**Comment by mortael** [comment]
I had the same issue on my Raspberry Pi running Python 3.8, after latest update.
I fixed it for myself by putting ssl_verify in config.ini to 1  (it was 0).

**Comment by jester-xbmc** [comment]
you beat me to it :) indeed fixes the issue on this side too

**Comment by Wally73** [comment]
works on windows as well thx for the fix @mortael

**Comment by BKSteve** [comment]
Renamed and reopened as solution. 
Closing others.

**Comment by mosscow** [comment]
I am not really that technical, took me ages to work out the config.ini was playing up. And I wiped a whole lot of settings to fix, reading the above the only setting i needed to fix was the ssl_verify

**Comment by peppermintesse** [comment]
I just updated using pip to sickchill-2022.8.15, and now when loading http://localhost:8081/home/ I get the following error. 

> **Error !
> 
> AttributeError: module 'sickchill.settings' has no attribute 'NO_LGMARGIN'**
> 
> `% endif </div> <div class="row"> % if settings.NO_LGMARGIN: <div class="col-lg-12 col-md-12 col-sm-12 col-xs-12"> % else: <div class="col-lg-10 col-lg-offset-1 col-md-10 col-md-offset-1 col-sm-12 col-xs-12"> % endif <%block name="content" />`
> 
> /Users/sandra/Library/Python/3.9/lib/python/site-packages/sickchill/gui/slick/views/layouts/main.mako, line 326:
> `% if settings.NO_LGMARGIN:`
> 
> /usr/local/lib/python3.9/site-packages/mako/runtime.py, line 940:
> `callable_(context, *args, **kwargs)`

in Terminal, it says:

`2022-08-16 09:24:19 INFO :: WEBSERVER-HOME_0 :: A mako error occurred: module 'sickchill.settings' has no attribute 'NO_LGMARGIN'`

I am not getting any messages re: `Cannot set verify_mode to CERT_NONE`, but I checked config.ini regardless, and `ssl_verify` is already set to 1. 

I'm using macOS 12.5, python 3.9. If this needs to be a different report, let me know.

**Comment by BKSteve** [comment]
@peppermintesse check `config.ini` under `[GUI]` there should be a `no_lgmargin = 1` and `sickchill/settings.py` line 281 should have `NO_LGMARGIN = False`
This is a new parameter but should be created new in `config.ini` at SC startup in after update.

**Comment by peppermintesse** [comment]
I checked `config.ini` and that file does **not** have that entry.

I found three different sickchill `settings.py` files. 

- One of them, modified today and under my home folder (/Users/sandra/Library/Python/3.9/lib/python/site-packages/sickchill), does have `NO_LGMARGIN = False` where you say it should be. 
- The one under /usr/local/lib/python3.9/site-packages/sickchill (dated Feb 21 2022) does not.
- The third, under /usr/local/lib/python3.8/site-packages/sickchill (dated Nov 13 2021) also does not.

I previously did an uninstall (`pip3 uninstall sickchill`) and reinstall (`pip3 install --user -U sickchill`) with no change. 

Thanks for your help. :)

**Comment by BKSteve** [comment]
Make sure your `user` has correct access to `config.ini` and your databases.
I'd suggest a full cleanup of the many SC installations. 
Backup (if you haven't already), uninstall, kill all SC folders in different pythons, reinstall SC, restore.

**Comment by peppermintesse** [comment]
> I'd suggest a full cleanup of the many SC installations.
Backup (if you haven't already), uninstall, kill all SC folders in different pythons, reinstall SC, restore.

I think that's going to have to be the way to go. :-/ Thanks again.

**Comment by idontknowwhatzowrite** [comment]
"ssl_verify = 1" in config.ini solved the issue for me

**Comment by valheru** [comment]
> I had the same issue on my Raspberry Pi running Python 3.8, after latest update. I fixed it for myself by putting ssl_verify in config.ini to 1 (it was 0).

Unfortunately this gives different issues, I now have my logs filling with this:

2022-08-19 12:26:34 INFO :: SEARCHQUEUE-BACKLOG-76656 :: [EliteTorrent] :: SSL Error requesting url: 'https://www.elitetorrent.eu/torrents.php?cat=4&modo=listado&orden=fecha&pag=1&buscar=the+tribe+5x24' Try disabling Cert Verification on the advanced tab of /config/general. Error: HTTPSConnectionPool(host='www.elitetorrent.eu', port=443): Max retries exceeded with url: /torrents.php?cat=4&modo=listado&orden=fecha&pag=1&buscar=the+tribe+5x24 (Caused by SSLError(SSLError(1, '[SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:997)')))
2022-08-19 12:26:32 INFO :: SEARCHQUEUE-BACKLOG-76656 :: [EliteTorrent] :: SSL Error requesting url: 'https://www.elitetorrent.eu/torrents.php?cat=4&modo=listado&orden=fecha&pag=1&buscar=the+tribe+5x46' Try disabling Cert Verification on the advanced tab of /config/general. Error: HTTPSConnectionPool(host='www.elitetorrent.eu', port=443): Max retries exceeded with url: /torrents.php?cat=4&modo=listado&orden=fecha&pag=1&buscar=the+tribe+5x46 (Caused by SSLError(SSLError(1, '[SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:997)')))
2022-08-19 12:26:30 INFO :: SEARCHQUEUE-BACKLOG-76656 :: [EliteTorrent] :: SSL Error requesting url: 'https://www.elitetorrent.eu/torrents.php?cat=4&modo=listado&orden=fecha&pag=1&buscar=The+Tribe+5x46' Try disabling Cert Verification on the advanced tab of /config/general. Error: HTTPSConnectionPool(host='www.elitetorrent.eu', port=443): Max retries exceeded with url: /torrents.php?cat=4&modo=listado&orden=fecha&pag=1&buscar=The+Tribe+5x46 (Caused by SSLError(SSLError(1, '[SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:997)')))
2022-08-19 12:26:28 INFO :: SEARCHQUEUE-BACKLOG-76656 :: [EliteTorrent] :: SSL Error requesting url: 'https://www.elitetorrent.eu/torrents.php?cat=4&modo=listado&orden=fecha&pag=1&buscar=the+tribe+5x45' Try disabling Cert Verification on the advanced tab of /config/general. Error: HTTPSConnectionPool(host='www.elitetorrent.eu', port=443): Max retries exceeded with url: /torrents.php?cat=4&modo=listado&orden=fecha&pag=1&buscar=the+tribe+5x45 (Caused by SSLError(SSLError(1, '[SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:997)')))
2022-08-19 12:26:26 INFO :: SEARCHQUEUE-BACKLOG-76656 :: [EliteTorrent] :: SSL Error requesting url: 'https://www.elitetorrent.eu/torrents.php?cat=4&modo=listado&orden=fecha&pag=1&buscar=The+Tribe+5x45' Try disabling Cert Verification on the advanced tab of /config/general. Error: HTTPSConnectionPool(host='www.elitetorrent.eu', port=443): Max retries exceeded with url: /torrents.php?cat=4&modo=listado&orden=fecha&pag=1&buscar=The+Tribe+5x45 (Caused by SSLError(SSLError(1, '[SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:997)')))

**Comment by BKSteve** [comment]
Set it back to `ssl_verify = 0` and see what happens.
If that resolves the `Error` and the prior error doesn't come back when you reenable the providers then great. 

Otherwise check this https://github.com/SickChill/SickChill/issues/8068#issuecomment-1216774206 to see if it can resolve the `CERT_NONE or NO_LGMARGIN` as an alternate solution.

**Comment by valheru** [comment]
If I set it to `ssl_verify = 0` sickchill won't start.
If I leave it at `ssl_verify = 1` I have issues with several torrent providers and my own Deluge server (Delug: authentication failed in the logs).

I am using Ubuntu 22.04 with Python 3.10 and Sickchill 2022.8.22.74751, from the interface:

SickChill Info:
Branch: master
Commit: cc9de09d40de432d8d86cd896ec424ab6af0d78d
Version: 2022.8.22.74751
Database Version: 44.5

Python Version:  3.10.4 (main, Jun 29 2022, 12:14:53) [GCC 11.2.0]
SSL Version:  OpenSSL 3.0.2 15 Mar 2022
OS:  Linux-5.15.0-40-generic-x86_64-with-glibc2.35
Locale:  en_US.UTF-8

**Comment by valheru** [comment]
With settings from precious comment (`ssl_verify = 1` and 'no_lgmargin = 0`) I get these errors in the logs:

AA ValueError: Cannot set verify_mode to CERT_NONE when check_hostname is enabled.
AA super(SSLContext, SSLContext).verify_mode.__set__(self, value)
AA File "/usr/lib/python3.10/ssl.py", line 737, in verify_mode
AA context.verify_mode = resolve_cert_reqs(self.cert_reqs)
AA File "/usr/local/lib/python3.10/dist-packages/urllib3/connection.py", line 349, in connect
AA conn.connect()
AA File "/usr/local/lib/python3.10/dist-packages/urllib3/connectionpool.py", line 978, in _validate_conn
AA self._validate_conn(conn)
AA File "/usr/local/lib/python3.10/dist-packages/urllib3/connectionpool.py", line 381, in _make_request
AA httplib_response = self._make_request(
AA File "/usr/local/lib/python3.10/dist-packages/urllib3/connectionpool.py", line 670, in urlopen
AA resp = conn.urlopen(
AA File "/usr/local/lib/python3.10/dist-packages/requests/adapters.py", line 489, in send
AA resp = super(CacheControlAdapter, self).send(request, **kw)
AA File "/usr/local/lib/python3.10/dist-packages/cachecontrol/adapter.py", line 53, in send
AA r = adapter.send(request, **kwargs)
AA File "/usr/local/lib/python3.10/dist-packages/requests/sessions.py", line 701, in send
AA resp = self.send(prep, **send_kwargs)
AA File "/usr/local/lib/python3.10/dist-packages/requests/sessions.py", line 587, in request
AA return self.request("GET", url, **kwargs)
AA File "/usr/local/lib/python3.10/dist-packages/requests/sessions.py", line 600, in get
AA self.response = self.session.get(self.url, timeout=120, verify=False)
AA File "/home/websites/sickchill/sickchill/oldbeard/clients/generic.py", line 239, in testAuthentication
AA result, message = client(host, username, password).testAuthentication()
AA File "/home/websites/sickchill/sickchill/views/home.py", line 235, in __torrent_test
AA return self.__torrent_test(host, username, password, torrent_method)
AA File "/home/websites/sickchill/sickchill/views/home.py", line 249, in testTorrent
AA return function()
AA File "/home/websites/sickchill/sickchill/views/index.py", line 187, in async_call
AA result = self.fn(*self.args, **self.kwargs)
AA File "/usr/lib/python3.10/concurrent/futures/thread.py", line 58, in run
AA value = future.result()
AA File "/usr/local/lib/python3.10/dist-packages/tornado/gen.py", line 762, in run
AA results = yield self.async_call(method)
AA File "/home/websites/sickchill/sickchill/views/index.py", line 161, in get
AA yielded = self.gen.throw(*exc_info)  # type: ignore
AA File "/usr/local/lib/python3.10/dist-packages/tornado/gen.py", line 769, in run
AA result = await result
AA File "/usr/local/lib/python3.10/dist-packages/tornado/web.py", line 1704, in _execute
AA Traceback (most recent call last):
AA During handling of the above exception, another exception occurred:
AA TypeError: Home.testTorrent() got an unexpected keyword argument 'torrent_method'
AA return function(**kwargs)
AA File "/home/websites/sickchill/sickchill/views/index.py", line 185, in async_call
AA Traceback (most recent call last):
AA HTTPServerRequest(protocol='http', host='10.0.1.1:8081', method='POST', uri='/home/testTorrent', version='HTTP/1.1', remote_ip='10.0.1.13')

**Comment by BKSteve** [comment]
First `no_lgmargin = 1` is only a mako setting to have or remove margins on large screens and is thrown as a tertiary error and not part of the SSL error.

Can you also add some actual SC log lines before and after the `AA` lines, it will help with assessing which CERT_NONE is kicking this.

Also put \``` that's 3 reverse dashes on the line above and line below the log. Helps with the layout.

One more thing, how are you installed? 
There's been updates to `master` which should have resolved this.
If pip then please do `pip3 install -U sickchill` as there's been some pip related issues too.

**Comment by valheru** [comment]
I installed via pip, tried your upgrade command which gave me this result:

```Installing collected packages: sickchill
  Attempting uninstall: sickchill
    Found existing installation: sickchill 2022.7.20
    Uninstalling sickchill-2022.7.20:
      Successfully uninstalled sickchill-2022.7.20
Successfully installed sickchill-2022.8.22
```

With `verify_ssl = 0` it still doesn't start, with `verify_ssl = 1` it starts, after starting I clicked on ` the test connection option in the interface under search settings -> torrent search which resulted in this in my logfile:

```
2022-08-28 00:04:40 INFO :: WEBSERVER :: Starting SickChill on http://10.0.1.1:8081/
2022-08-28 00:04:58 ERROR :: WEBSERVER :: Uncaught exception POST /home/testTorrent (10.0.1.13)
HTTPServerRequest(protocol='http', host='10.0.1.1:8081', method='POST', uri='/home/testTorrent', version='HTTP/1.1', remote_ip='1
0.0.1.13')
Traceback (most recent call last):
  File "/home/websites/sickchill/sickchill/views/index.py", line 185, in async_call
    return function(**kwargs)
TypeError: Home.testTorrent() got an unexpected keyword argument 'torrent_method'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/tornado/web.py", line 1704, in _execute
    result = await result
  File "/usr/local/lib/python3.10/dist-packages/tornado/gen.py", line 769, in run
    yielded = self.gen.throw(*exc_info)  # type: ignore
  File "/home/websites/sickchill/sickchill/views/index.py", line 161, in get
    results = yield self.async_call(method)
  File "/usr/local/lib/python3.10/dist-packages/tornado/gen.py", line 762, in run
    value = future.result()
  File "/usr/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/websites/sickchill/sickchill/views/index.py", line 187, in async_call
    return function()
  File "/home/websites/sickchill/sickchill/views/home.py", line 249, in testTorrent
    return self.__torrent_test(host, username, password, torrent_method)
  File "/home/websites/sickchill/sickchill/views/home.py", line 235, in __torrent_test
    result, message = client(host, username, password).testAuthentication()
  File "/home/websites/sickchill/sickchill/oldbeard/clients/generic.py", line 239, in testAuthentication
    self.response = self.session.get(self.url, timeout=120, verify=False)
  File "/usr/local/lib/python3.10/dist-packages/requests/sessions.py", line 600, in get
    return self.request("GET", url, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/requests/sessions.py", line 587, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.10/dist-packages/requests/sessions.py", line 701, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/cachecontrol/adapter.py", line 53, in send
    resp = super(CacheControlAdapter, self).send(request, **kw)
  File "/usr/local/lib/python3.10/dist-packages/requests/adapters.py", line 489, in send
    resp = conn.urlopen(
  File "/usr/local/lib/python3.10/dist-packages/urllib3/connectionpool.py", line 670, in urlopen
    httplib_response = self._make_request(
  File "/usr/local/lib/python3.10/dist-packages/urllib3/connectionpool.py", line 381, in _make_request
    self._validate_conn(conn)
  File "/usr/local/lib/python3.10/dist-packages/urllib3/connectionpool.py", line 978, in _validate_conn
    conn.connect()
  File "/usr/local/lib/python3.10/dist-packages/urllib3/connection.py", line 349, in connect
    context.verify_mode = resolve_cert_reqs(self.cert_reqs)
  File "/usr/lib/python3.10/ssl.py", line 737, in verify_mode
    super(SSLContext, SSLContext).verify_mode.__set__(self, value)
ValueError: Cannot set verify_mode to CERT_NONE when check_hostname is enabled.
```

**Comment by BKSteve** [comment]
@valheru this issue is actively being worked on. Your error relates to `providers` which is being actively worked on in #8099.

Link to open invite https://github.com/SickChill/SickChill/issues/8055#issuecomment-1229097659 for anyone interested in helping out. We just can't test everything every time, and fixing one thing can often causes others.

**Comment by valheru** [comment]
Ok, thank you. I am happy to know this is being worked on.

**Comment by miigotu** [comment]
If anyone comes across this, it is fixed in master, and will be in 2022.08.30 (or the next release if I don't get to it today).
